AnnService/inc/Helper/StringConvert.h

```diff
@@ -95,7 +95,7 @@ inline bool ConvertStringToUnsignedInt(const char* p_str, DataType& p_value)
         return false;
     }
 
-    if (val < (std::numeric_limits<DataType>::min)() || val >(std::numeric_limits<DataType>::max)())
```

Comment:
yes, so it will check whether the uint64_t we get will overflow or not. When it will be negative? This function will only be called when the DataType is uint8_t, uint16_t, uint32_t and uint64_t in the function ConvertStringTo<XXX>.

Reply:
I mean if the string is negative

Sample of the behavior I think doesn't make sense
Code:
```C++
template <typename DataType>
void TestUnsignedConv(const std::string & s)
{
  DataType v = 0;
  int a = 0;
  bool success = ConvertStringToUnsignedInt<DataType>(s.c_str(), v);
  std::cout << "Converted  \""<< s << "\" to value " << static_cast<unsigned int>(v) << ". Success: " << success << std::endl;
}

int main() {
  TestUnsignedConv<std::uint8_t>("128");
  TestUnsignedConv<std::uint8_t>("257");
  TestUnsignedConv<std::uint8_t>("-1");
  TestUnsignedConv<std::uint8_t>("-18446744073709551600");
}
```
Output:
```
Converted  "128" to value 128. Success: 1
Converted  "257" to value 0. Success: 0
Converted  "-1" to value 0. Success: 0
Converted  "-18446744073709551600" to value 16. Success: 1
```
It is because `std::strtoull` will do the overflow, if it overflows into range the result is unintuitive, since it overflows based on the limits of `std::uint64_t` instead of `DataType`.